### DISCUSSION TITLE
Unit test: detecting an original fixture file path

### DISCUSSION THREAD

**Discussion by AlexeyOnsite** [rectorphp/rector#6245] [Q&A]
Hello,
first of all thank you for this great tool :)

I'm developing a fairly complex rule to automatically convert old spagetti code into Symfony controller (similar to your inspirational article "How to Migrate Spaghetti to 304 Symfony 5 Controllers Over Weekend" I guess).

The rule depends on original file path to create the destination path like:
/foo/bar/test_snakecase/my_script.php -> /src/Controller/Legacy/TestSnakecase/MyScriptController.php

The problem is, \Rector\Testing\PHPUnit\AbstractRectorTestCase::doTestFileInfo uses StaticFixtureSplitter::splitFileInfoToLocalInputAndExpectedFileInfos which creates files in the temp directory. The part of the original path (i. e. test_snakecase) is lost.

What can I do to solve this problem?

**Comment by TomasVotruba** [comment]
Hi,

we're aware of this and welcome any improvement.
Currently you can compare the end of file paths, not the directories.

You can find some inspiration in this rule, that checks a new file was added:
https://github.com/rectorphp/rector/blob/baa26986f2ca4bd10cb0746983551cf4e82050d9/rules-tests/Autodiscovery/Rector/Class_/MoveServicesBySuffixToDirectoryRector/MoveServicesBySuffixToDirectoryRectorTest.php#L20-L21

**Reply by AlexeyOnsite** [reply]
Yes, I've already took a look at this.
One possible solution coud be to make AbstractRectorTestCase::doTestFileMatchesExpectedContent protected or do I miss something?

**Reply by TomasVotruba** [reply]
Depends on your use case. Give it a try and if it works, we can change it in core :+1:

**Reply by AlexeyOnsite** [reply]
It works!
